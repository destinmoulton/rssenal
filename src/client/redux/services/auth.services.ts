import ky from "../../lib/ky";

import { API_AUTH_LOGIN, API_AUTH_VALIDATE_TOKEN } from "../apiendpoints";

import {
    generateJWTJSONHeaders,
    generateJSONHeaders,
    generateJWTHeaders
} from "../../lib/headers";
import * as Types from "../../types";

export async function apiValidateToken() {
    const url = API_AUTH_VALIDATE_TOKEN;
    const options = {
        headers: generateJWTHeaders()
    };

    return ky
        .get(url, options)
        .then(res => {
            if (!res.ok) {
                return { status: "invalid" };
            }

            return res.json();
        })
        .then(auth => {
            if (auth.status === "valid") {
                return true;
            } else {
                throw new Error("User not logged in.");
            }
        })
        .catch(err => {
            throw err;
        });
}

export async function apiLoginUser(username: string, password: string) {
    const url = API_AUTH_LOGIN;
    const options = {
        json: { username, password },
        headers: generateJSONHeaders()
    };

    return ky
        .post(url, options)
        .then(res => {
            return res.json();
        })
        .then(auth => {
            if (auth.status === "success") {
                return auth.token;
            } else {
                throw new Error("Invalid username or password.");
            }
        })
        .catch(err => {
            throw err;
        });
}
