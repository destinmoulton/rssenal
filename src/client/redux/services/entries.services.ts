import * as moment from "moment";

import { API_ENTRIES_BASE } from "../apiendpoints";
import { generateJWTJSONHeaders, generateJWTHeaders } from "../../lib/headers";
import * as Types from "../../types";

export function apiGetEntriesForFeed(
    feedId: Types.TFeedID,
    shouldShowRead: boolean
) {
    console.log("running apiGetEntriesForFeed", feedId);
    const queryString = `?showEntriesHasRead=${shouldShowRead}&feedId=${feedId}`;
    const url = API_ENTRIES_BASE + queryString;
    const init = {
        method: "GET",
        headers: generateJWTHeaders()
    };

    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(resObj => {
            if (resObj.status === "error") {
                throw new Error(resObj.error);
            } else if (Array.isArray(resObj.entries)) {
                return resObj.entries;
            }
        })
        .catch(err => {
            throw err;
        });
}

export function apiUpdateEntryHasRead(entry: Types.IEntry, hasRead: boolean) {
    const url = API_ENTRIES_BASE + entry._id;
    const init = {
        method: "PUT",
        body: JSON.stringify({ has_read: hasRead }),
        headers: generateJWTJSONHeaders()
    };

    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(resObj => {
            if (resObj.status === "error") {
                throw new Error(resObj.error);
            } else {
                return true;
            }
        })
        .catch(err => {
            throw err;
        });
}

export function ammendRawEntries(
    currentFeed: Types.IFeed,
    rawEntries: Types.IEntry[]
): Types.IEntry[] {
    return rawEntries.map((entry: Types.IEntry) => {
        const feedTitle = currentFeed.title;
        const timeAgo = moment(entry.publish_date).fromNow();
        return {
            ...entry,
            feedTitle,
            timeAgo
        };
    });
}

export function addAmmendedEntries(
    currentEntries: Types.TEntries,
    ammendedEntries: Types.IEntry[]
) {
    let fullEntries = currentEntries.toOrderedMap();
    ammendedEntries.forEach(entry => {
        fullEntries = fullEntries.set(entry._id, entry);
    });
    return fullEntries;
}

export function ammendEntryReadStatus(
    currentEntries: Types.TEntries,
    entryId: Types.TEntryID,
    hasRead: boolean
): Types.TEntries {
    const entry = currentEntries.get(entryId);
    const newEntry = { ...entry, has_read: hasRead };
    return currentEntries.set(entryId, newEntry);
}
