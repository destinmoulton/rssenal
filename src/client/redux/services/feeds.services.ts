import { OrderedMap } from "immutable";

import { API_FEEDS_BASE } from "../apiendpoints";
import { generateJWTJSONHeaders, generateJWTHeaders } from "../../lib/headers";
import { propertyComparator } from "../../lib/sort";
import * as Types from "../../types";

export async function apiAddFeed(feedInfo: Types.IFeed) {
    const url = API_FEEDS_BASE;
    const headers = generateJWTJSONHeaders();
    const init = {
        method: "POST",
        body: JSON.stringify({ ...feedInfo }),
        headers
    };

    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(feedObj => {
            if (feedObj.status === "error") {
                throw new Error(feedObj.error);
            } else if (feedObj.status === "success") {
                return feedObj.feedInfo;
            }
        })
        .catch(err => {
            throw err;
        });
}

export async function apiDeleteFeed(feedID: Types.TFeedID) {
    const url = API_FEEDS_BASE + feedID;
    const init = {
        method: "DELETE",
        headers: generateJWTHeaders()
    };
    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(resObj => {
            if (resObj.status === "error") {
                throw new Error(resObj.error);
            } else if (resObj.status === "success") {
                return true;
            }
        })
        .catch(err => {
            throw err;
        });
}

export async function apiGetAllFeeds() {
    const url = API_FEEDS_BASE;
    const init = {
        method: "GET",
        headers: generateJWTHeaders()
    };
    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(resObj => {
            if (resObj.status === "error") {
                throw new Error(resObj.error);
            } else if (resObj.status === "success") {
                return resObj.feeds;
            }
        })
        .catch(err => {
            throw err;
        });
}

export async function apiUpdateFeed(feedInfo: Types.IFeed) {
    const url = API_FEEDS_BASE + feedInfo._id;
    const init = {
        method: "PUT",
        body: JSON.stringify({
            title: feedInfo.title,
            folder_id: feedInfo.folder_id
        }),
        headers: generateJWTJSONHeaders()
    };

    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(feedObj => {
            if (feedObj.status === "error") {
                throw new Error(feedObj.error);
            } else if (feedObj.status === "success") {
                return feedObj.feedInfo;
            }
        })
        .catch(err => {
            throw err;
        });
}

export function convertRawFeedsToOrderedMap(
    feedsArr: Types.IFeed[]
): Types.TFeeds {
    const feedsTuples = feedsArr.map(feed => {
        return [feed._id, feed];
    });
    return OrderedMap(feedsTuples);
}

export function updateUnreadCount(
    unreadMap: Types.IFeedsUnreadMap,
    entries: Types.TEntries,
    feeds: Types.TFeeds
) {
    let entriesCounted = unreadMap.entriesCounted;
    let unreadFeeds = unreadMap.feeds;
    let unreadFolders = unreadMap.folders;

    entries.map((entry: Types.IEntry) => {
        if (!entriesCounted.has(entry._id) && !entry.has_read) {
            entriesCounted = entriesCounted.add(entry._id);

            const feed = feeds.get(entry.feed_id);

            if (!unreadFeeds.has(feed._id)) {
                unreadFeeds = unreadFeeds.set(feed._id, 1);
            } else {
                const countUnread: number = unreadFeeds.get(feed._id);
                unreadFeeds = unreadFeeds.set(feed._id, countUnread + 1);
            }

            if (!unreadFolders.has(feed.folder_id)) {
                unreadFolders = unreadFolders.set(feed.folder_id, 1);
            } else {
                const countUnread: number = unreadFolders.get(feed.folder_id);
                unreadFolders = unreadFolders.set(
                    feed.folder_id,
                    countUnread + 1
                );
            }
        }
    });

    return {
        entriesCounted,
        feeds: unreadFeeds,
        folders: unreadFolders
    };
}

export function decrementUnread(
    feedId: Types.TFeedID,
    feeds: Types.TFeeds,
    unreadMap: Types.IFeedsUnreadMap
) {
    const feed = feeds.get(feedId);

    let unreadFeeds = unreadMap.feeds;
    let unreadFolders = unreadMap.folders;

    if (unreadFeeds.has(feedId)) {
        const unreadFeedCount = unreadFeeds.get(feedId);
        const newCount = unreadFeedCount > 1 ? unreadFeedCount - 1 : 0;
        unreadFeeds = unreadFeeds.set(feedId, newCount);
    }

    if (unreadFolders.has(feed.folder_id)) {
        const unreadFolderCount = unreadFolders.get(feed.folder_id);
        const newCount = unreadFolderCount > 1 ? unreadFolderCount - 1 : 0;
        unreadFolders = unreadFolders.set(feed.folder_id, newCount);
    }

    return {
        ...unreadMap,
        feeds: unreadFeeds,
        folders: unreadFolders
    };
}

export function setSingleFeed(feed: Types.IFeed, feeds: Types.TFeeds) {
    return feeds.set(feed._id, feed);
}

export function sortFeeds(feeds: Types.TFeeds): Types.TFeeds {
    return feeds
        .sort((a: Types.IFeed, b: Types.IFeed) =>
            propertyComparator(a, b, "asc", "title", true)
        )
        .toOrderedMap();
}
