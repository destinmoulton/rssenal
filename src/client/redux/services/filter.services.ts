import * as Types from "../../types";

export function filterVisibleEntries(
    filter: Types.IFilter,
    currentEntries: Types.TEntries,
    feeds: Types.TFeeds
) {
    let title = "All";

    let filteredEntries = currentEntries.toOrderedMap();
    switch (filter.limit) {
        case "feed":
            const activeFeed = feeds.get(filter.id);
            title = activeFeed.title;

            filteredEntries = currentEntries
                .filter(entry => {
                    return entry.feed_id === filter.id;
                })
                .toOrderedMap();
            break;
        case "folder":
            if (filter.id !== "all") {
                const activeFolder = folders.get(filter.id);

                title = activeFolder.name;

                const feedIds = feeds
                    .filter(feed => {
                        return feed.folder_id === filter.id;
                    })
                    .map(feed => {
                        return feed._id;
                    });

                filteredEntries = currentEntries
                    .filter(entry => {
                        return feedIds.includes(entry.feed_id);
                    })
                    .toOrderedMap();
            }
            break;
    }

    return { title, filteredEntries };
}
