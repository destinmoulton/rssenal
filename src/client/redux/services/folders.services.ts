import { OrderedMap } from "immutable";
import { API_FOLDERS_BASE, API_FOLDERS_GET_ALL } from "../apiendpoints";
import { UNCATEGORIZED_FOLDER } from "../../constants";
import { generateJWTJSONHeaders, generateJWTHeaders } from "../../lib/headers";

import * as Types from "../../types";

export async function apiAddFolder(newFolderName: string) {
    const url = API_FOLDERS_BASE;
    const init = {
        method: "POST",
        body: JSON.stringify({ name: newFolderName }),
        headers: generateJWTJSONHeaders()
    };
    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(resObj => {
            if (resObj.status === "error") {
                throw new Error("Failed to add folder.");
            } else {
                return resObj;
            }
        })
        .catch(err => {
            throw err;
        });
}

export async function apiDeleteFolder(folderID: Types.TFolderID) {
    const url = API_FOLDERS_BASE + folderID;
    const init = {
        method: "DELETE",
        headers: generateJWTHeaders()
    };
    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(resObj => {
            if (resObj.status === "success") {
                return true;
            } else {
                throw new Error("Failed to remove folder.");
            }
        })
        .catch(err => {
            throw err;
        });
}

export async function apiGetAllFolders() {
    const url = API_FOLDERS_GET_ALL;
    const init = {
        method: "GET",
        headers: generateJWTHeaders()
    };
    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(folders => {
            return folders;
        })
        .catch(err => {
            throw err;
        });
}

export async function apiReorderFolders(foldersArr: Types.IFolder[]) {
    const url = API_FOLDERS_BASE;
    const init = {
        method: "PUT",
        body: JSON.stringify({ folders: foldersArr }),
        headers: generateJWTJSONHeaders()
    };
    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(resObj => {
            if (resObj.status === "success") {
                return true;
            } else {
                throw new Error("Failed to reorder folders.");
            }
        })
        .catch(err => {
            throw err;
        });
}

export async function apiSaveFolder(folderInfo: Types.IFolder) {
    const url = API_FOLDERS_BASE + folderInfo._id;
    const init = {
        method: "PUT",
        body: JSON.stringify({ name: folderInfo.name }),
        headers: generateJWTJSONHeaders()
    };

    return fetch(url, init)
        .then(res => {
            return res.json();
        })
        .then(resObj => {
            if (resObj.status === "error") {
                throw new Error("Failed to save folder.");
            } else {
                return resObj;
            }
        })
        .catch(err => {
            throw err;
        });
}

export function convertRawFoldersToOrderedMap(
    foldersArr: Types.IFolder[]
): Types.TFolders {
    const arrayMap = foldersArr.map(folder => {
        return [folder._id, folder];
    });
    const mappedFolders: Types.TFolders = OrderedMap(arrayMap);
    return mappedFolders.set(UNCATEGORIZED_FOLDER._id, UNCATEGORIZED_FOLDER);
}

export function setFolder(
    folder: Types.IFolder,
    folders: Types.TFolders
): Types.TFolders {
    return folders.set(folder._id, folder);
}

export function removeFolderFromOrderedMap(
    folderID: Types.TFolderID,
    folders: Types.TFolders
) {
    return folders.delete(folderID);
}
