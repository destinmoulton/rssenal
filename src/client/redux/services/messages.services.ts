import * as Types from "../../types";

interface IMessageInfo {
    messageText: string;
    level: any;
}
export function addMessage(
    messageInfo: IMessageInfo,
    nextUID: number,
    messages: Types.TMessages
) {
    const message: Types.IMessage = {
        message: messageInfo.messageText,
        level: messageInfo.level,
        uid: nextUID
    };

    return messages.push(message);
}

export function getNextUID(lastUID: number) {
    return lastUID + 1;
}

export function removeMessage(
    messageToRemove: Types.IMessage,
    messages: Types.TMessages
) {
    const index = messages.findIndex((message: Types.IMessage) => {
        return message.uid === messageToRemove.uid;
    });

    return messages.remove(index);
}
