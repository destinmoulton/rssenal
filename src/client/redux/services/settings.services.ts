import * as Types from "../../types";

export function addSetting(
    settings: Types.TSettings,
    setting_key: string,
    setting_value: any
) {
    const setting = Object.assign({}, settings.get(setting_key));

    setting.value = setting_value;
    return settings.set(setting_key, setting);
}
