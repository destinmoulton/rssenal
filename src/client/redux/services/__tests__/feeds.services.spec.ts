import * as fetchMock from "fetch-mock";

import { Map, Set, OrderedMap } from "immutable";

import * as FeedsServices from "../feeds.services";

import FEED from "../../../../../test/data/feed";
import API_FEEDS_STRING from "../../../../../test/data/api/feeds";
import IMM_FEEDS from "../../../../../test/data/immutable/feeds";
import IMM_ENTRIES from "../../../../../test/data/immutable/ammendedEntries";

describe("feeds.services", () => {
    afterEach(() => {
        fetchMock.reset();
        fetchMock.restore();
        localStorage.clear();
    });

    describe("apiAddFeed()", () => {
        it("throws an error for an error response", async () => {
            const url = "/api/feeds/";
            fetchMock.postOnce(url, { body: { status: "error" } });

            expect.assertions(2);

            try {
                await FeedsServices.apiAddFeed(FEED);
            } catch (err) {
                expect(err).toBeInstanceOf(Error);

                await fetchMock.flush();
                expect(fetchMock.done()).toBe(true);
            }
        });

        it("returns the new feed for a valid response", async () => {
            const expectedFeed = { feedData: "FEED DATA HERE" };
            const URL = "/api/feeds/";
            fetchMock.postOnce(URL, {
                body: {
                    status: "success",
                    feedInfo: expectedFeed
                }
            });

            expect.assertions(2);

            try {
                const feedInfo = await FeedsServices.apiAddFeed(FEED);
                expect(feedInfo).toEqual(expectedFeed);

                await fetchMock.flush();
                expect(fetchMock.done()).toBe(true);
            } catch (err) {
                throw err;
            }
        });
    });

    describe("apiDeleteFeed()", () => {
        it("throws an error for an error response", async () => {
            const url = "/api/feeds/FEED_ID";
            fetchMock.deleteOnce(url, { body: { status: "error" } });

            expect.assertions(2);

            try {
                await FeedsServices.apiDeleteFeed("FEED_ID");
            } catch (err) {
                expect(err).toBeInstanceOf(Error);
                expect(fetchMock.done()).toBe(true);
            }
        });

        it("returns the new feed for a valid response", async () => {
            const expectedFeed = { feedData: "FEED DATA HERE" };
            const url = "/api/feeds/FEED_ID";
            fetchMock.deleteOnce(url, {
                body: {
                    status: "success"
                }
            });

            expect.assertions(2);

            try {
                const res = await FeedsServices.apiDeleteFeed("FEED_ID");
                expect(res).toBe(true);

                await fetchMock.flush();
                expect(fetchMock.done()).toBe(true);
            } catch (err) {
                throw err;
            }
        });
    });

    describe("apiGetAllFeeds()", () => {
        it("throws an error for an error response", async () => {
            const url = "/api/feeds/";
            fetchMock.getOnce(url, { status: "error" });

            expect.assertions(2);

            try {
                await FeedsServices.apiGetAllFeeds();
            } catch (err) {
                expect(err).toBeInstanceOf(Error);
                expect(fetchMock.done()).toBe(true);
            }
        });

        it("returns the feeds for a valid response", async () => {
            const expectedFeeds = ["feed1", "feed2"];
            const url = "/api/feeds/";
            fetchMock.getOnce(url, {
                status: "success",
                feeds: expectedFeeds
            });

            expect.assertions(2);

            try {
                const res = await FeedsServices.apiGetAllFeeds();
                expect(res).toEqual(expectedFeeds);

                await fetchMock.flush();
                expect(fetchMock.done()).toBe(true);
            } catch (err) {
                throw err;
            }
        });
    });

    describe("apiUpdateFeed()", () => {
        it("throws an error for an error response", async () => {
            const feedID = "5b33c76cb2438d5708dc197e";
            const url = "/api/feeds/" + feedID;
            fetchMock.putOnce(url, { body: { status: "error" } });

            expect.assertions(2);

            try {
                await FeedsServices.apiUpdateFeed(FEED);
            } catch (err) {
                expect(err).toBeInstanceOf(Error);
                expect(fetchMock.done()).toBe(true);
            }
        });

        it("returns the updated feed for a valid response", async () => {
            const expectedFeed = { feed: "UPDATED FEED" };
            const feedID = "5b33c76cb2438d5708dc197e";
            const url = "/api/feeds/" + feedID;
            fetchMock.putOnce(url, {
                body: {
                    status: "success",
                    feedInfo: expectedFeed
                }
            });

            expect.assertions(2);

            try {
                const res = await FeedsServices.apiUpdateFeed(FEED);
                expect(res).toEqual(expectedFeed);

                await fetchMock.flush();
                expect(fetchMock.done()).toBe(true);
            } catch (err) {
                throw err;
            }
        });
    });

    describe("apiValidateFeedURL()", () => {
        it("throws an error for an error response", async () => {
            const url = "/api/validatefeedurl/";
            fetchMock.postOnce(url, { body: { status: "error" } });

            expect.assertions(2);

            try {
                await FeedsServices.apiValidateFeedURL("TESTURL");
            } catch (err) {
                expect(err).toBeInstanceOf(Error);

                await fetchMock.flush();
                expect(fetchMock.done()).toBe(true);
            }
        });

        it("returns the validated feed for a valid response", async () => {
            const expectedFeed = { feedData: "FEED DATA HERE" };
            const url = "/api/validatefeedurl/";
            fetchMock.postOnce(url, {
                body: {
                    status: "success",
                    feedInfo: expectedFeed
                }
            });

            expect.assertions(2);

            try {
                const feedInfo = await FeedsServices.apiValidateFeedURL(
                    "TESTURL"
                );
                expect(feedInfo).toEqual(expectedFeed);

                await fetchMock.flush();
                expect(fetchMock.done()).toBe(true);
            } catch (err) {
                throw err;
            }
        });
    });

    it("convertRawFeedsToOrderedMap() returns an OrderedMap<> of feeds", () => {
        const om = FeedsServices.convertRawFeedsToOrderedMap(
            JSON.parse(API_FEEDS_STRING).feeds
        );
        expect(om).toEqual(IMM_FEEDS);
    });

    it("updateUnreadCont() creates a new unread count; matches snapshot", () => {
        const initUnreadMap = {
            entriesCounted: Set<string>(),
            feeds: Map<string, number>(),
            folders: Map<string, number>()
        };
        const newUnreadCount = FeedsServices.updateUnreadCount(
            initUnreadMap,
            IMM_ENTRIES,
            IMM_FEEDS
        );

        expect(newUnreadCount).toMatchSnapshot();
    });

    it("decrementUnreadCount() decreases the read count for a feed; matches snapshot", () => {
        const feedID = "5b33c76cb2438d5708dc197e";
        const initUnreadMap = {
            entriesCounted: Set<string>(),
            feeds: Map<string, number>(),
            folders: Map<string, number>()
        };
        const initUnreadCount = FeedsServices.updateUnreadCount(
            initUnreadMap,
            IMM_ENTRIES,
            IMM_FEEDS
        );

        const updatedUnreadCount = FeedsServices.decrementUnread(
            feedID,
            IMM_FEEDS,
            initUnreadCount
        );

        expect(updatedUnreadCount).toMatchSnapshot();
    });

    it("setSingleFeed() sets a feed in the OrderedMap of feeds", () => {
        const feedID = "5b33c76cb2438d5708dc197e";
        const CLONED_FEED = { ...FEED, title: "NEW TITLE" };
        const newFeedsOM = FeedsServices.setSingleFeed(CLONED_FEED, IMM_FEEDS);
        expect(newFeedsOM).toEqual(OrderedMap([[feedID, CLONED_FEED]]));
    });

    it("sortFeeds() sorts feeds in asc titular order", () => {
        const unorderedFeeds = [
            { ...FEED, _id: "BID", title: "b9Second" },
            { ...FEED, _id: "AID", title: "a9First" },
            { ...FEED, _id: "0ID", title: "09Num" }
        ];

        const orderedFeeds = [
            { ...FEED, _id: "0ID", title: "09Num" },
            { ...FEED, _id: "AID", title: "a9First" },
            { ...FEED, _id: "BID", title: "b9Second" }
        ];

        const unorderedOM = FeedsServices.convertRawFeedsToOrderedMap(
            unorderedFeeds
        );
        const expectedOM = FeedsServices.convertRawFeedsToOrderedMap(
            orderedFeeds
        );
        const sorted = FeedsServices.sortFeeds(unorderedOM);
        expect(sorted).toEqual(expectedOM);
    });
});
