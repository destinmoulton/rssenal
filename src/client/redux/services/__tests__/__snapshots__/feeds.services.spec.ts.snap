// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feeds.services decrementUnreadCount() decreases the read count for a feed; matches snapshot 1`] = `
Object {
  "entriesCounted": Immutable.Set [
    "5b33c76cb2438d5708dc197f",
    "5b33c76cb2438d5708dc1981",
    "5b33c76cb2438d5708dc1983",
    "5b33c76cb2438d5708dc1985",
    "5b33c76cb2438d5708dc1987",
  ],
  "feeds": Immutable.Map {
    "5b33c76cb2438d5708dc197e": 4,
  },
  "folders": Immutable.Map {
    "5b33c57eb2438d5708dc1968": 4,
  },
}
`;

exports[`feeds.services updateUnreadCont() creates a new unread count; matches snapshot 1`] = `
Object {
  "entriesCounted": Immutable.Set [
    "5b33c76cb2438d5708dc197f",
    "5b33c76cb2438d5708dc1981",
    "5b33c76cb2438d5708dc1983",
    "5b33c76cb2438d5708dc1985",
    "5b33c76cb2438d5708dc1987",
  ],
  "feeds": Immutable.Map {
    "5b33c76cb2438d5708dc197e": 5,
  },
  "folders": Immutable.Map {
    "5b33c57eb2438d5708dc1968": 5,
  },
}
`;
